### PR TITLE
Update dependencies and Flask syntax

### PR DESCRIPTION
**Pull request by mikellewade**
Updates Flask syntax/dependencies to reflect changes in Unit 2

https://app.asana.com/0/1205655776549324/1206697443678563

### CODE REVIEW

File: app/routes.py [hunk 1]
```diff
@@ -1,40 +1,41 @@
...
-    return jsonify(response.json())
+    return response.json(), 200
 
-@proxy_bp.route("/weather", methods=["GET"])
+@bp.get("/weather")
 def get_weather():
     lat_query = request.args.get("lat")
     lon_query = request.args.get("lon")
     print(lat_query, lon_query)
+    print(weather_key)
```
Comment (kelsey-steven-ada) [inline app/routes.py]: How do you feel about removing these? I feel a little weird about leaving print statements in our routes, but I don't feel super strongly if there's a reason to leave them in.

File: requirements.txt [hunk 2]
```diff
@@ -1,35 +1,22 @@
...
+Mako==1.3.2
+MarkupSafe==2.1.5
+packaging==23.2
+pluggy==1.4.0
+psycopg2-binary==2.9.9
+pytest==8.0.0
+python-dotenv==1.0.1
+SQLAlchemy==2.0.25
+typing_extensions==4.9.0
+Werkzeug==3.0.1
```
Comment (kelsey-steven-ada) [inline requirements.txt]: It looks like we're missing the `requests` package & its dependencies in requirements.txt. When I tried to run the server I got the error:
```py
  File "/Users/kelseysteven/Documents/curriculum_changes/weather-report-proxy-server/app/routes.py", line 5, in <module>
    import requests
ModuleNotFoundError: No module named 'requests'
```

Once I did `pip install requests`, I was able to run the server & hit both the endpoints as expected.

```suggestion
alembic==1.13.1
blinker==1.7.0
certifi==2024.8.30
charset-normalizer==3.4.0
click==8.1.7
Flask==3.0.2
Flask-Cors==5.0.0
Flask-Migrate==4.0.5
Flask-SQLAlchemy==3.1.1
greenlet==3.0.3
gunicorn==21.2.0
idna==3.10
iniconfig==2.0.0
itsdangerous==2.1.2
Jinja2==3.1.3
Mako==1.3.2
MarkupSafe==2.1.5
packaging==23.2
pluggy==1.4.0
psycopg2-binary==2.9.9
pytest==8.0.0
python-dotenv==1.0.1
requests==2.32.3
SQLAlchemy==2.0.25
typing_extensions==4.9.0
urllib3==2.2.3
Werkzeug==3.0.1
```

**Review by kelsey-steven-ada — CHANGES_REQUESTED** [review]
Thank you for handling these Flask updates with the Vite React Weather Report work! 🌟 We need to add the `requests` package to the requirements, everything else is looking good!

File: app/routes.py (hunk 1 above)
Comment (mikellewade) [reply app/routes.py]: Oh! I left the second one in there by accident (the key wasn't working but it just took a second to activate). I can remove it! The first one I am okay with removing as well. I wasn't quite sure why they were there, I figured they were for the students to check the coordinates, but that is something they can do themselves. I can remove it.

**Review by kelsey-steven-ada — APPROVED** [review]

**Pull request merged**